loopflow-computation/src/main/java/com/farao_community/farao/loopflow_computation/LoopFlowComputationImpl.java

```diff
@@ -56,17 +63,33 @@ public LoopFlowResult calculateLoopFlows(Network network, SensitivityAnalysisPar
     @Override
     public LoopFlowResult buildLoopFlowsFromReferenceFlowAndPtdf(SystematicSensitivityResult alreadyCalculatedPtdfAndFlows, Set<FlowCnec> flowCnecs) {
         LoopFlowResult results = new LoopFlowResult();
-
         for (FlowCnec flowCnec : flowCnecs) {
             double refFlow = alreadyCalculatedPtdfAndFlows.getReferenceFlow(flowCnec);
-            double commercialFLow = getGlskStream(flowCnec)
+            double commercialFLow = getGlskStream(flowCnec).filter(entry -> isInMainComponent(entry.getValue(), network))
                 .mapToDouble(entry -> alreadyCalculatedPtdfAndFlows.getSensitivityOnFlow(entry.getValue(), flowCnec) * referenceProgram.getGlobalNetPosition(entry.getKey()))
                 .sum();
             results.addCnecResult(flowCnec, refFlow - commercialFLow, commercialFLow, refFlow);
         }
         return results;
     }
 
+    static boolean isInMainComponent(LinearGlsk linearGlsk, Network network) {
+        for (String glsk : linearGlsk.getGLSKs().keySet()) {
+            Generator generator = network.getGenerator(glsk);
+            if (generator != null) {
+                if (generator.getTerminal().getBusView().getBus().isInMainConnectedComponent()) {
+                    return true;
+                }
+            } else {
+                Load load = network.getLoad(glsk);
+                if (load != null && load.getTerminal().getBusView().getBus().isInMainConnectedComponent()) {
+                    return true;
+                }
+            }
+        }
+        return false;
```

Comment:
i would prefer to throw an exception in that case, as it is not suppose to happen and this case can "hide" some other problems in the previous steps.

Reply:
it would throw an exception even if all loads / gens exist, but are not connected to the main component ?

Reply:
I was more thinking : 
if the load and the generator does not exist -> throw an exception (that case should be spotted in the GLSKs methods)
if one of them exist and is disconnected -> return false

Reply:
alright ! i updated it with a random commit message :p

Reply:
:)